File: android-lib/logcat-logger/src/main/java/com/example/toysapp/logcat_logger/LogcatLogger.java
package com.example.toysapp.logcat_logger;

import android.util.Log;

import com.example.toysapp.logger.Logger;

import org.jetbrains.annotations.NotNull;

public final class LogcatLogger implements Logger {

    private static final String TAG = "LogcatLogger";

    @Override
    public void log(@NotNull String message) {
        Log.d(TAG, "Log from LogcatLogger: " + message);
    }
}

File: android-lib/logcat-logger/src/main/java/com/example/toysapp/logcat_logger/LogcatLogger.kt
package com.example.toysapp.logcat_logger

import android.util.Log
import com.example.toysapp.logger.Logger

class LogcatLogger: Logger {

    override fun log(message: String) {
        Log.d(TAG, "Log from LogcatLogger: $message")
    }

    companion object {
        private const val TAG = "LogcatLogger"
    }
}
